glusterd2/main.go

```diff
@@ -119,6 +120,36 @@ func main() {
 		log.WithError(err).Fatal("Failed to generate local auth token")
 	}
 
+	// Get the Jaeger endpoints from the command line if provided
+	jaegerEndpoints := make(map[string]string)
+	jaegerEndpoint, _ := flag.CommandLine.GetString(tracing.JaegerEndpointOpt)
+	jaegerAgentEndpoint, _ := flag.CommandLine.GetString(tracing.JaegerAgentEndpointOpt)
+
+	// Create the Opencensus Jaeger exporter with info from command line
+	exporter := tracing.InitJaegerExporter(jaegerEndpoint, jaegerAgentEndpoint)
+	if exporter != nil {
+		defer exporter.Flush()
+		jaegerEndpoints[tracing.JaegerEndpointOpt] = jaegerEndpoint
+		jaegerEndpoints[tracing.JaegerAgentEndpointOpt] = jaegerAgentEndpoint
+		// Register the Jaeger exporter
+		tracing.RegisterJaegerExporter(exporter, jaegerEndpoints)
+	}
+
+	// Get the Jaeger endpoints from config file if provided
+	jaegerEndpoint2 := config.GetString(tracing.JaegerEndpointOpt)
```

Comment:
Just this second initialization from the values obtained from `config` package is enough. `config.Get*` functions fetch configured with the following priority, CLI > Config file > default values. You do not need to do a seperate fetch.

Reply:
This makes things much simpler! Thanks for pointing this out. I saw the way logging related options were being reinitialized and followed it.

Reply:
Logging is a special case because. We need logging to be available before we read-in the config file.